front/pages/w/[wId]/u/chat/[cId]/index.tsx

```diff
@@ -629,18 +653,40 @@ export default function AppChat({
     setInput(input);
   };
 
-  const handleSwitchDataSourceSelection = (name: string) => {
+  const handleSwitchDataSourceSelection = async (name: string) => {
+    let selected = false;
+
     const newSelection = dataSources.map((ds) => {
       if (ds.name === name) {
+        selected = !ds.selected;
         return {
           ...ds,
-          selected: !ds.selected,
+          selected,
         };
       } else {
         return ds;
       }
     });
     setDataSources(newSelection);
+    try {
+      const currentStickySelectionRaw = await getUserMetadataFromClient(
```

Comment:
this is the part that kind of sucks, we make 2 API calls each time the DS selection changes.

Reply:
Let's update on submit